ruzzle_bare_minimum.py

```diff
@@ -49,72 +52,85 @@ def gen_graph():
     for x in range(4):
         for y in range(4):
             for cx, cy in directions:
-                if 0 <= x+cx < 4 and 0 <= y+cy < 4:
-                    graph[(x, y)].append((x+cx, y+cy))
+                if 0 <= x + cx < 4 and 0 <= y + cy < 4:
+                    graph[(x, y)].append((x + cx, y + cy))
+
+    return graph
 
-    return graph    
 
 def dfs(graph, visited, s, word, word_pts, word_mult, path):
     """Start at point s and search for words, keeping track of the word, points, and multiplier"""
-    global words_info, points, word_mults
-    if 2 <= len(word) <= 6: # ex) if a word begins with TTH, this doesn't exist so we can stop adding letters
-        if word not in prefixes[len(word)-2]:
+    global words_info, points, word_int_mults
+    len_word = len(word)
+
+    if 2 <= len_word <= 6:  # ex) if a word begins with TTH, this doesn't exist so we can stop adding letters
+        if word not in prefixes[len_word - 2]:
             return
-    if len(word) >= 2: # store all 2 letter possible words in words_info (may not be an actual word)
+
+    # store all >2 letter possible words in words_info if they are actual words
+    if len_word >= 2 and word.lower() in dictionary:
         score = word_pts * word_mult
-        bonus = 5 * (len(word) - 4) # length bonus 
-        bonus = max(0, bonus) # bonus can't be negative
-        words_info.append((word, score + bonus, path[:])) # append copy of path 
-    if len(word) == 12: # there are no words greater than 12 letters (based on ruzzle database), so stop searching
+        bonus = 0 if len_word < 4 else 5 * (len_word - 4)  # length bonus
+        words_info.append((word, score + bonus, path[:]))  # append copy of path
+
+    # there are no words greater than 12 letters (based on ruzzle database), so stop searching
+    if len_word == 12:
         return
-    visited[s] = True # begin DFS, make sure no overlaps
-    for x in graph[s]: # graph[s] contains the adjacent points
-        if not visited[x]:
-            visited[x] = True
-            word += board[x[0]][x[1]] # add letter to word
-            word_pts += points[x[0]][x[1]] # add points
-            if word_mults[x[0]][x[1]] in '23': # if multiplier, set multiplier
-                word_mult *= int(word_mults[x[0]][x[1]])
-            path.append(x) # add position to path list
-            dfs(graph, visited, x, word, word_pts, word_mult, path) # search from this new point
-            visited[x] = False # reset everything to continue to search in other directions
-            word = word[:-1]
-            word_pts -= points[x[0]][x[1]]
-            if word_mults[x[0]][x[1]] in '23':
-                word_mult //= int(word_mults[x[0]][x[1]])
-            del path[-1]
-
-def all_combos(board):
+
+    visited[s] = True  # begin DFS, make sure no overlaps
+    path.append(None)
```

Comment:
What does path.append(None) do?

Reply:
Instead of appending to the path each time, I append once at the start, set the last element to the specific vertex, and then delete the vertex at the end.

Reply:
That's quite smart. I'm going to look over the rest of the code and I'll merge it soon. Are all the double spaces part of PEP8? Like the double space before inline comment, double blank line between functions, etc.

Reply:
Yes - all of the formatting changes (aside from a bit of spacing for readability inside functions) are PEP8.